packages/fuels-contract/src/contract.rs

```diff
@@ -341,7 +341,7 @@ impl Contract {
     fn should_compute_call_data_offset(args: &[Token]) -> bool {
         match args
             .iter()
-            .any(|t| matches!(t, Token::Struct(_) | Token::Enum(_)))
+            .any(|t| matches!(t, Token::Struct(_) | Token::Enum(_) | Token::B256(_)))
```

Comment:
Tuples should probably be on this list as well? Then there's probably the question about strings and arrays.

Maybe it's better if we invert the condition to check for copy types since they are a limited well defined set that will likely never change?

Reply:
We don't support tuples: https://github.com/FuelLabs/fuels-rs/blob/0c5b9f2c8b89745b1f29e7d8fc15e834e066967d/packages/fuels-core/src/lib.rs#L103-L115

> Maybe it's better if we invert the condition to check for copy types since they are a limited well defined set that will likely never change?

That's a good idea, but this whole thing will go away very soon, as soon as [this is done](https://github.com/FuelLabs/fuels-rs/issues/201). So I didn't wanna shake things too much, but I'm not married to this idea. What do you think?

Reply:
Ah of course, you're right... let's keep it as is :)

Reply:
I agree with Rodrigo as well